pycode/memilio-surrogatemodel/memilio/surrogatemodel/ode_secir_simple/model.py

```diff
@@ -0,0 +1,190 @@
+    @param max_epochs int maximum number of epochs in training. 
+    @param early_stop int defines the number of epochs without significant change tolerated before forcing an early stop of training. 
+
+    """
+
+    if not os.path.isfile(os.path.join(path, 'data_secir_simple.pickle')):
+        ValueError("no dataset found in path: " + path)
+
+    file = open(os.path.join(path, 'data_secir_simple.pickle'), 'rb')
+
+    data = pickle.load(file)
+    data_splitted = data_generation.splitdata(data["inputs"], data["labels"])
+
+    train_inputs = data_splitted["train_inputs"]
+    train_labels = data_splitted["train_labels"]
+    valid_inputs = data_splitted["valid_inputs"]
+    valid_labels = data_splitted["valid_labels"]
+    test_inputs = data_splitted["test_inputs"]
+    test_labels = data_splitted["test_labels"]
+
+    early_stopping = tf.keras.callbacks.EarlyStopping(monitor='val_loss',
+                                                      patience=early_stop,
+                                                      mode='min')
+
+    model.compile(
+        loss=tf.keras.losses.MeanSquaredError(),
+        optimizer=tf.keras.optimizers.Adam(),
+        metrics=[tf.keras.metrics.MeanAbsoluteError()])
+
+    history = model.fit(train_inputs, train_labels, epochs=max_epochs,
+                        validation_data=(valid_inputs, valid_labels),
+                        callbacks=[early_stopping])
+
+    if(plot):
+        plot_losses(history)
+        plotCol(test_inputs, test_labels, model=model,
+                plot_col='InfectedSymptoms', max_subplots=6)
+        df = get_test_statistic(test_inputs, test_labels, model)
+        print(df)
+    return history
+
+
+def plot_losses(history):
+    """! Plots the losses of the model training.  
+
+    @param history model training history. 
+
+    """
+    plt.plot(history.history['loss'])
+    plt.plot(history.history['val_loss'])
+    plt.title('model loss')
+    plt.ylabel('loss')
+    plt.xlabel('epoch')
+    plt.legend(['train', 'val'], loc='upper left')
+    if os.path.isdir("plots") == False:
+        os.mkdir("plots")
+    plt.savefig('plots/losses_plot.pdf')
+    plt.show()
+
+
+def get_test_statistic(test_inputs, test_labels, model):
+    """! Calculates the mean absolute percentage error based on the test dataset.   
+
+    @param test_inputs inputs from test data.
+    @param testlabels labels/outout from test data.
+    @aram model trained model. 
+
+    """
+
+    pred = model(test_inputs)
+    pred = pred.numpy()
+    test_labels = np.array(test_labels)
+
+    diff = pred - test_labels
+    anteil = (abs(diff))/abs(test_labels)
+
+    anteil_6 = anteil.transpose(2, 0, 1).reshape(6, -1)
+    df = pd.DataFrame(data=anteil_6)
+    df = df.transpose()
+
+    mean_percentage = pd.DataFrame(
+        data=(df.mean().values) * 100,
+        index=['Exposed', 'Carrier', 'Infected', 'Hospitalized', 'ICU',
+               'Dead'],
+        columns=['Percentage Error'])
```

Comment:
Do you really need a new dataframe here? You could also rename columns of df and multiply it by 100?